src/Sentry.Maui/Internal/SentryMauiScreenshotProcessor.cs

```diff
@@ -18,6 +18,11 @@ public SentryMauiScreenshotProcessor(SentryMauiOptions options)
 
     public SentryEvent? Process(SentryEvent @event, SentryHint hint)
     {
+        if (!_options.BeforeCaptureInternal?.Invoke(@event, hint) ?? false)
+        {
+            return @event;
+        }
+
```

Comment:
Probably the most meaningful code to test in this PR would be this bit... so an additional couple of `SentryMauiScreenshotProcessorTest` tests. 

Basically we want to ensure screenshots do get captured if the BeforeCapture callback hasn't been set or returns true... and that a screenshot doesn't get captured if it's set and returns false.

Reply:
> if the BeforeCapture callback hasn't been set

We don't have a test for this. Only tests with callback set

Reply:
I'll add this now

Reply:
however, `CaptureException_WhenAttachScreenshots_ContainsScreenshotAttachmentAsync` is already testing when BeforeCapture is not set